config/configcompression/compressiontype.go

```diff
@@ -25,19 +37,46 @@ func (ct *Type) IsCompressed() bool {
 	return *ct != typeEmpty && *ct != typeNone
 }
 
-func (ct *Type) UnmarshalText(in []byte) error {
-	typ := Type(in)
-	if typ == TypeGzip ||
-		typ == TypeZlib ||
-		typ == TypeDeflate ||
-		typ == TypeSnappy ||
-		typ == TypeZstd ||
-		typ == TypeLz4 ||
-		typ == typeNone ||
-		typ == typeEmpty {
-		*ct = typ
+func (ct *TypeWithLevel) UnmarshalText(in []byte) error {
+	var err error
+	parts := strings.Split(string(in), "/")
+	compressionTyp := Type(parts[0])
+	level := zlib.DefaultCompression
+	if len(parts) == 2 {
+		level, err = strconv.Atoi(parts[1])
+		if err != nil {
+			return fmt.Errorf("invalid compression level: %q", parts[1])
+		}
+		if compressionTyp == TypeSnappy ||
+			compressionTyp == TypeLz4 ||
+			compressionTyp == typeNone ||
+			compressionTyp == typeEmpty {
+			return fmt.Errorf("compression level is not supported for %q", compressionTyp)
+		}
+	}
+	ct.Level = Level(level)
+	if (compressionTyp == TypeGzip && isValidLevel(level)) ||
+		(compressionTyp == TypeZlib && isValidLevel(level)) ||
+		(compressionTyp == TypeDeflate && isValidLevel(level)) ||
+		compressionTyp == TypeSnappy ||
+		compressionTyp == TypeLz4 ||
+		compressionTyp == TypeZstd ||
```

Comment:
Seems that we're supporting specifying a level for zstd -- should there be validation for the level here?

Reply:
zstd has this function which returns an encoder level that closest matches the compression ratio of a specific zstd compression level. So I didn't see a need to specify a range of valid levels for it.

This function (its used here) - https://pkg.go.dev/github.com/klauspost/compress/zstd#EncoderLevelFromZstd

Reply:
OK. It's somewhat tying to the implementation details of klauspost/compress/zstd, but I suppose it's fine.